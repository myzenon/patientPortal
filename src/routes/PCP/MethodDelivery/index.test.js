import React from 'react'
import { MethodDelivery } from './index'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

test('renders correctly', () => {
    const props = {
        cmr: {
            ...mockObj.cmr,
        },
    }
    const wrapper = render(
        <MethodDelivery  { ...props } />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test('clicks Next without checking the checkbox | go to final screen page', (done) => {
    const props = {
        cmr: {
            ...mockObj.cmr,
        }, callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirCmrIdPut')
            expect(get).toEqual({
                id: mockObj.cmr.id
            })
            expect(post).toEqual({
                deliveryMethods: []
            })
            return Promise.resolve()
        },
        setCMR: (cmr) => {
            expect(cmr).toEqual({
                ...mockObj.cmr,
                deliveryMethods: []
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/final-screen')
            done()
        }
    }
    const wrapper = mount(
        <MethodDelivery  { ...props } />
    )
    setTimeout(() => {
        wrapper.find('button').first().simulate('click')
    }, 0)
})


test('clicks Next with checking the checkbox | go to final screen page', (done) => {
    const props = {
        cmr: {
            ...mockObj.cmr,
        }, callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirCmrIdPut')
            expect(get).toEqual({
                id: mockObj.cmr.id
            })
            expect(post).toEqual({
                deliveryMethods: [{ code: "mail" }]
            })
            return Promise.resolve()
        },
        setCMR: (cmr) => {
            expect(cmr).toEqual({
                ...mockObj.cmr,
                deliveryMethods: [{ code: "mail" }]
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/final-screen')
            done()
        }
    }
    const wrapper = mount(
        <MethodDelivery  { ...props } />
    )
    setTimeout(() => {
        wrapper.find('input[type="checkbox"]').first().simulate('click')
        wrapper.find('button').first().simulate('click')
    }, 0)
})