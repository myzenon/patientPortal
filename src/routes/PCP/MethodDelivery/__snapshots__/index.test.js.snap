// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <p>
      <strong>
        Once our pharmacist completes your medication review,  We will email a link to your personal medication list and medication action plan.
      </strong>
    </p>
    <table>
      <tbody>
        <tr
          class="checkboxWrapper"
        >
          <td>
            <input
              class="checkbox"
              name="getByEmail"
              type="checkbox"
            />
          </td>
          <td>
            <span>
              In addition to the email, I would like a hard copy mailed to me.
            </span>
          </td>
        </tr>
      </tbody>
    </table>
    <button
      class="button btn btn-secondary"
      type="button"
    >
      Next
    </button>
  </div>
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepCompleted"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="stepCompleted"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumberActive"
          >
            2
          </div>
        </div>
        <div
          class="stepCompleted"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumberActive"
          >
            3
          </div>
        </div>
        <div
          class="stepCompleted"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumberActive"
          >
            4
          </div>
        </div>
        <div
          class="stepActive"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumberActive"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
